smithy-swift-codegen/src/test/kotlin/waiters/WaiterAcceptorGeneratorTests.kt

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package waiters
+
+import MockHttpRestJsonProtocolGenerator
+import TestContext
+import defaultSettings
+import getFileContents
+import io.kotest.matchers.string.shouldContainOnlyOnce
+import org.junit.jupiter.api.Test
+import software.amazon.smithy.codegen.core.SymbolProvider
+import software.amazon.smithy.model.Model
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.model.shapes.ServiceShape
+import software.amazon.smithy.swift.codegen.SwiftSettings
+import software.amazon.smithy.swift.codegen.core.CodegenContext
+import software.amazon.smithy.swift.codegen.integration.ProtocolGenerator
+import software.amazon.smithy.swift.codegen.integration.SwiftIntegration
+import software.amazon.smithy.swift.codegen.model.expectShape
+import software.amazon.smithy.swift.codegen.waiters.WaiterAcceptorGenerator
+import software.amazon.smithy.waiters.WaitableTrait
+
+class WaiterAcceptorGeneratorTests {
```

Comment:
It'd be nice to have a few more tests covering the variants of acceptors, such as those with mapping and flat mapping and compact mapping etc.

Reply:
We already have functional tests for these operations in the other repo.  I added tests there for projection this morning as a result of a build failure on one AWS service.

Really the only thing that we gain with these tests when we have the other ones is coverage of the generated comments.

All these tests do is break when I make superficial changes to the acceptor code, so if I add these it will be the very last thing before I merge.

Reply:
>Really the only thing that we gain with these tests when we have the other ones is coverage of the generated comments.

We also gain better PR reviews since we easily see the generated code alongside the PR. Additionally, we get test failures that are easier to debug. If a tests fails during codegen, then its obvious that we aren't generating the expected code. If a protocol tests fails, that doesn't necessarily mean we're generating the wrong code but rather that the behavior is unexpected. The fix might be to update the generated code or it might be to update swift code etc. It's really nice to have the failure point narrowed down.

Reply:
This it non-blocking btw, just think it'd be nice to have.